client/lib/polyfills.js

```diff
@@ -0,0 +1,96 @@
+function promisePolyfill () {
+  if (!window.Promise) {
+    window.Promise = require('promise');
```

Comment:
this will _always_ require the `promise` library. _right_...?
What we need is to _conditionally_ load the library as a script from the CDN rather than bundling it.